src/client/splitor.cpp

```diff
@@ -387,7 +387,7 @@ bool Splitor::MarkDiscardBitmap(IOTracker* iotracker, FileSegment* fileSegment,
     fileSegment->SetBitmap(offset, len);
 
     if (fileSegment->IsAllBitSet()) {
-        iotracker->discardSegments_.push_back(segmentIndex);
+        iotracker->discardSegments_.emplace(segmentIndex);
```

Comment:
Repeat push segmentIndex before?

Reply:
Yes, if a segment is discarded, internal bitmap is all set, the condition here `fileSegment->IsAllBitSet()` is true.
The internal bitmap will be cleared by reading/writing operations or segment has been discarded, e.g., DeAllocateSegment RPC to MDS succeeded.
If RPC hasn't sent to MDS, and the same segment is been discarded again(can be triggered by `fstrim`), this condition will be always true.
Because the logic here is processing a chunk, so it will push segment_size/chunk_size same segment index.